xtevent/_eventols.ado

```diff
@@ -193,6 +194,20 @@ program define _eventols, rclass
 	
 	gen byte `esample' = e(sample)
 	
+	* DiD estimate 
+	
+	loc lwindow = abs(`lwindow')
+	loc rwindow `rwindow'
+	if "`diffavg'"!=""{
+		di as text _n "DiD estimate and standard error from lincom:"
+		#d;
+		lincom  ((_b[ _k_eq_p0] + _b[ _k_eq_p1] + _b[ _k_eq_p2] + _b[ _k_eq_p3] 
+			+ _b[ _k_eq_p4] + _b[ _k_eq_p5] + _b[ _k_eq_p6]) / (`rwindow' + 2)) 
+			- ((_b[_k_eq_m6] + _b[_k_eq_m5] + _b[_k_eq_m4] + _b[_k_eq_m3] + 
+			_b[_k_eq_m2]) / (`lwindow' + 1)), cformat(%9.4g);
+		#d cr
+	}
+	

```

Comment:
@rayhuang11 This solution is not general yet! This would only work if the estimation window is 5. Can you generalize to arbitrary windows?
@Constantino-Carreto-Romero  pitch in to help please if @rayhuang11 has questions.

Reply:
@jorpppp thanks for the catch. That's on me, I totally missed that!